### ISSUE TITLE
Updating grades

### ISSUE THREAD

**Issue by Celso1912** [packbackbooks/lti-1-3-php-library#73]
Is there any way to update the scores from all userid(students) in all resourceIds given a role. For instance: a teacher uses the tool to update his students' grades for all asssigned activities. Currently I'm able to update the score for one particular resourceId at a time only.

thanks

**Comment by dbhynds** [comment]
I don't believe so. Their documentation specifies that a `userId` must be present in the payload to update a score for a line item: https://www.imsglobal.org/spec/lti-ags/v2p0/#userid

You'll need to send updated scores individually. A word of warning with that: you should not make these requests in parallel. Canvas will rate-limit you make too many concurrent requests. You'll need to do them one at a time.
